packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -0,0 +1,220 @@
+import { DISPLAYNAME_PREFIX } from "../../common/props";
+import { Button, IButtonProps } from "../button/buttons";
+import { Dialog, IDialogProps } from "./dialog";
+import { IStepProps, Step, StepId } from "./step";
+
+type StepElement = React.ReactElement<IStepProps & { children: React.ReactNode }>;
+
+export interface IMultistepDialogProps extends IDialogProps {
+    /**
+     * Whether to disable the next button in the panel footer.
+     */
+    disableNext?: boolean;
+
+    /**
+     * Props for the button to display on the final step.
+     */
+    finalButtonProps?: Partial<IButtonProps>;
+
+    /**
+     * A callback that is invoked when the user selects a different step by clicking on back, next, or a step itself.
+     */
+    onChange?(newStepId: StepId, prevStepId: StepId | undefined, event: React.MouseEvent<HTMLElement>): void;
+}
+
+interface IMultistepDialogState {
+    lastViewedIndex: number;
+    selectedIndex: number;
+}
+
+const PADDING_BOTTOM = 0;
+
+const MIN_WIDTH = 800;
+
+const INITIAL_STATE = {
+    lastViewedIndex: 0,
+    selectedIndex: 0,
+};
+
+@polyfill
+export class MultistepDialog extends AbstractPureComponent2<IMultistepDialogProps, IMultistepDialogState> {
+    public static displayName = `${DISPLAYNAME_PREFIX}.MultistepDialog`;
+
+    public static defaultProps: Partial<IMultistepDialogProps> = {
+        canOutsideClickClose: true,
+        isOpen: false,
+    };
+
+    public constructor(props: IMultistepDialogProps) {
+        super(props);
+        this.state = INITIAL_STATE;
+    }
+
+    public render() {
+        return (
+            <Dialog {...this.props} style={this.getDialogStyle()}>
+                <div className={Classes.MULTISTEP_DIALOG_PANELS}>
+                    {this.renderLeftPanel()}
+                    {this.maybeRenderRightPanel()}
+                </div>
+            </Dialog>
+        );
+    }
+
+    public componentDidUpdate(prevProps: IMultistepDialogProps) {
+        if (!prevProps.isOpen && this.props.isOpen) {
+            this.setState(INITIAL_STATE);
+        }
+    }
+
+    private getDialogStyle() {
+        return { paddingBottom: PADDING_BOTTOM, minWidth: MIN_WIDTH, ...this.props.style };
+    }
+
+    private renderLeftPanel() {
+        return (
+            <div className={Classes.MULTISTEP_DIALOG_LEFT_PANEL}>
+                {this.getStepChildren().filter(isStepElement).map(this.renderStep)}
+            </div>
+        );
+    }
+
+    private renderStep = (step: StepElement, index: number) => {
+        const stepNumber = index + 1;
+        const hasBeenViewed = this.state.lastViewedIndex >= index;
+        const currentlySelected = this.state.selectedIndex === index;
+        return (
+            <div className={classNames(Classes.STEP_CONTAINER, { [Classes.ACTIVE]: hasBeenViewed })} key={index}>
+                <div
+                    className={classNames(Classes.STEP, { [Classes.ACTIVE]: hasBeenViewed })}
```

Comment:
do we need all 4 active classes? I think we can remove this one and the one on L113 and change the Sass to be something like:

```sass
.step {
  .active & {
    ...
  }
}
```